app/src/androidTest/java/org/simple/clinic/overdue/communication/CommunicationRepositoryAndroidTest.kt

```diff
@@ -0,0 +1,61 @@
+package org.simple.clinic.overdue.communication
+
+import com.google.common.truth.Truth
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.simple.clinic.AppDatabase
+import org.simple.clinic.TestClinicApp
+import org.simple.clinic.TestData
+import org.simple.clinic.login.LoginResult
+import org.simple.clinic.patient.SyncStatus
+import org.simple.clinic.user.UserSession
+import java.util.UUID
+import javax.inject.Inject
+
+class CommunicationRepositoryAndroidTest {
+
+  @Inject
+  lateinit var repository: CommunicationRepository
+
+  @Inject
+  lateinit var database: AppDatabase
+
+  @Inject
+  lateinit var userSession: UserSession
+
+  @Inject
+  lateinit var testData: TestData
+
+  @Before
+  fun setup() {
+    TestClinicApp.appComponent().inject(this)
+
+    val loginResult = userSession.saveOngoingLoginEntry(testData.qaOngoingLoginEntry())
+        .andThen(userSession.loginWithOtp(testData.qaUserOtp()))
+        .blockingGet()
+    Truth.assertThat(loginResult).isInstanceOf(LoginResult.Success::class.java)
+  }
+
+  @Test
+  fun when_creating_new_communication_then_the_communication_should_be_saved() {
+    val appointmentId = UUID.randomUUID()
+
+    repository
+        .create(appointmentId, Communication.Type.MANUAL_CALL, result = Communication.Result.AGREED_TO_VISIT)
+        .blockingAwait()
+
+    val savedCommunication = repository.pendingSyncRecords().blockingGet()[0]
+    savedCommunication.apply {
+      Truth.assertThat(this.appointmentId).isEqualTo(appointmentId)
```

Comment:
Why asserting individual properties instead of the data class?

Reply:
Oh wow I'm feeling so stupid. Will fix this.

Reply:
Update: not so stupid afterall